PHP-FPM could crash when timeout is configured

### Relevant telegraf.conf

```toml
[[inputs.phpfpm]]
  urls = ["fcgi://127.0.0.1:9008/status"]
  timeout = "5s"
```


### Logs from Telegraf

```text
$ ./telegraf -test -config telegraf.conf 
2024-04-17T09:23:12Z I! Loading config: telegraf.conf
2024-04-17T09:23:12Z I! Starting Telegraf unknown brought to you by InfluxData the makers of InfluxDB
2024-04-17T09:23:12Z I! Available plugins: 234 inputs, 9 aggregators, 32 processors, 25 parsers, 60 outputs, 5 secret-stores
2024-04-17T09:23:12Z I! Loaded inputs: phpfpm
2024-04-17T09:23:12Z I! Loaded aggregators: 
2024-04-17T09:23:12Z I! Loaded processors: 
2024-04-17T09:23:12Z I! Loaded secretstores: 
2024-04-17T09:23:12Z W! Outputs are not used in testing mode!
2024-04-17T09:23:12Z I! Tags enabled: host=MBP-de-pierre
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x38 pc=0x1053632f4]

goroutine 116 [running]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.newFcgiClient(0x12a05f200, {0x14001bc23c8?, 0x9?}, {0x14000de6708?, 0x0?, 0x0?})
        /Users/pierref/dev/telegraf/plugins/inputs/phpfpm/fcgi_client.go:36 +0x1f4
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0x14000edaa00, {0x14001bc23c1, 0x1c}, {0x10b7c8d40, 0x14000ee62a0})
        /Users/pierref/dev/telegraf/plugins/inputs/phpfpm/phpfpm.go:162 +0x24c
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1({0x14001bc23c1?, 0x0?})
        /Users/pierref/dev/telegraf/plugins/inputs/phpfpm/phpfpm.go:129 +0x70
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather in goroutine 115
        /Users/pierref/dev/telegraf/plugins/inputs/phpfpm/phpfpm.go:127 +0x68
```


### System info

Telegraf master, MacOS 14.4

### Docker

_No response_

### Steps to reproduce

1. Do NOT run php-fpm, have nothing listening on port 9008
2. Run Telegraf --test



### Expected behavior

Don't crash :)

### Actual behavior

Crash

### Additional info

🤦 I've caused this issue in #15036 because I've never tested with no php-fpm running.

I'll submit a PR to fix in the next hours.

--- a/plugins/inputs/phpfpm/fcgi_client.go
+++ b/plugins/inputs/phpfpm/fcgi_client.go
@@ -29,7 +29,10 @@ func newFcgiClient(timeout time.Duration, h string, args ...interface{}) (*conn,
 		laddr := net.UnixAddr{Name: args[0].(string), Net: h}
 		con, err = net.DialUnix(h, nil, &laddr)
 	default:
-		err = errors.New("fcgi: we only accept int (port) or string (socket) params")
+		return nil, errors.New("fcgi: we only accept int (port) or string (socket) params")
+	}
+	if err != nil {
+		return nil, err
 	}
 
 	if timeout != 0 {
@@ -38,11 +41,7 @@ func newFcgiClient(timeout time.Duration, h string, args ...interface{}) (*conn,
 		}
 	}
 
-	fcgi := &conn{
-		rwc: con,
-	}
-
-	return fcgi, err
+	return &conn{rwc: con}, nil
 }
 
 func (c *conn) Request(env map[string]string, requestData string) (retout []byte, reterr []byte, err error) {

--- a/plugins/inputs/phpfpm/phpfpm_test.go
+++ b/plugins/inputs/phpfpm/phpfpm_test.go
@@ -187,6 +187,35 @@ func TestPhpFpmTimeout_From_Fcgi(t *testing.T) {
 	require.GreaterOrEqual(t, time.Since(start), timeout)
 }
 
+// TestPhpFpmCrashWithTimeout_From_Fcgi show issue #15175: when timeout is enabled
+// and nothing is listenning on specified port, a nil pointer was dereferenced.
+func TestPhpFpmCrashWithTimeout_From_Fcgi(t *testing.T) {
+	tcp, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err, "Cannot initialize test server")
+
+	tcpAddress := tcp.Addr().String()
+
+	// Yes close the tcp port now. The listenner is only used to find a free
+	// port and then make it free. This test hope that nothing will re-use the
+	// port in meantime.
+	tcp.Close()
+
+	const timeout = 200 * time.Millisecond
+
+	//Now we tested again above server
+	r := &phpfpm{
+		Urls:    []string{"fcgi://" + tcpAddress + "/status"},
+		Timeout: config.Duration(timeout),
+		Log:     &testutil.Logger{},
+	}
+	require.NoError(t, r.Init())
+
+	var acc testutil.Accumulator
+	require.Error(t, acc.GatherError(r.Gather))
+
+	require.Empty(t, acc.GetTelegrafMetrics())
+}
+
 func TestPhpFpmGeneratesMetrics_From_Socket(t *testing.T) {
 	// Create a socket in /tmp because we always have write permission and if the
 	// removing of socket fail when system restart /tmp is clear so

